stackslib/src/chainstate/burn/operations/leader_key_register.rs

```diff
@@ -77,6 +77,12 @@ impl LeaderKeyRegisterOp {
         Some(LeaderKeyRegisterOp::new(&prover_pubk))
     }
 
+    /// Interpret the first 20 bytes of the key registration's memo field as the Hash160 of
+    ///  of the public key that will sign this miner's nakamoto blocks.
+    pub fn interpret_nakamoto_signing_key(&self) -> Option<Hash160> {
```

Comment:
Ah, interesting idea!  This should get added to the SIP and design doc.  Can you add a quick one-sentence comment in both places so we remember to merge it in?  Thanks!

Reply:
Also, this field should probably be explicit in `LeaderKeyRegisterOp` as an `Option<..>`.  The field would be required to be `Some(..)` in epoch 3.0, but it's not considered in earlier epochs.

Reply:
I think there is a little bit of code clarity tradeoff with adding a new field to the struct, and I'm genuinely not sure which would work out more elegantly. If there's a new field in the struct, there's two kind of immediate impacts:

1. The db schema of the Sortition DB needs to change and FromRow/ToSql need to handle the new field. The db schema of the burnchain DB doesn't change, but the deserialization of the opcode would need to make sure that it handles the absence of the field correctly. *Or* the schema of the SortitionDB doesn't change and the FromRow / Insert method handles shuttling the new field via the existing memo column.
2. The consensus deserialize/serialize needs to be sure to be compatible. This is pretty straight forward and I'm not worried about its impact.

In light of (1), do you still think there should be a new field for this struct?

Reply:
I think that in each breaking change, we've required users to spin up from genesis anyway, so introducing a new field which renders the Stacks 2.x chainstate incompatible I think is fine (especially if it simplifies the consensus code).

Reply:
I can take this on